src/metabase/native_query_analyzer/impl.cljc

```diff
@@ -41,6 +43,14 @@
      ;; For both MySQL and SQL Server, whether identifiers are case-sensitive depends on database configuration only,
      ;; and quoting has no effect on this, so we disable this option for consistency with `:case-insensitive`.
      :quotes-preserve-case? (not (contains? #{:mysql :sqlserver} driver))
+     :features              {:postgres-syntax        (isa? driver/hierarchy driver :postgres)
+                             :square-bracket-quotes  (= :sqlserver driver)
```

Comment:
In theory I suppose there could be a driver that descends from `:sqlserver`, but in practice I don't think there is and I hope that the shape of this all changes soon anyway.